superset/assets/src/visualizations/deckgl/CategoricalDeckGLContainer.jsx

```diff
@@ -48,19 +48,20 @@ export default class CategoricalDeckGLContainer extends React.PureComponent {
    */
 
   /* eslint-disable-next-line react/sort-comp */
-  static getDerivedStateFromProps(nextProps) {
-    const fd = nextProps.slice.formData;
+  static getDerivedStateFromProps(nextProps, currentState) {
+    const fd = nextProps.payload.form_data;
 
     const timeGrain = fd.time_grain_sqla || fd.granularity || 'PT1M';
     const timestamps = nextProps.payload.data.features.map(f => f.__timestamp);
     const { start, end, getStep, values, disabled } = getPlaySliderParams(timestamps, timeGrain);
-    const categories = getCategories(fd, nextProps.payload.data.features);
+    const categories = currentState.categories ||
```

Comment:
Umm, with the current code. After this `currentState.categories` is set once, that value will stay same until component unmount and never replaced by `getCategories(...)`. 

This will not work when the dataset changes, new filters are applied and categories change, etc.

Reply:
Yeah, I knew I was doing something stupid. Let me fix the code. Thanks, @kristw!